xapian-core/configure.ac

```diff
@@ -279,6 +279,28 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
 #endif
 ]])], [msvc=no], [msvc=yes])
 
+XAPIAN_CMAKE_SHARED=
+if test "$enable_shared" = yes; then
+  XAPIAN_CMAKE_SHARED=ON
+else
+  XAPIAN_CMAKE_SHARED=OFF
+fi
+AC_SUBST([XAPIAN_CMAKE_SHARED])
+
+if test "$msvc" = yes; then
+    LIBRARY_STATIC_EXT=.lib
+else
+    LIBRARY_STATIC_EXT=.a
+fi
+AC_SUBST([LIBRARY_STATIC_EXT])
```

Comment:
Rather than hard-coding assumptions based on platform or compiler, I'd suggest `LIBRARY_STATIC_EXT=$libext` which is the variable libtool sets to the extension it decides to use (this variable is documented in the libtool manual, not some internal implementation detail).

Reply:
Actually:

```
LIBRARY_STATIC_EXT=.$libext 
```

The `.` isn't included in this variable.